src/IlkRegistry.sol

```diff
@@ -137,10 +152,23 @@ contract IlkRegistry {
         (address _pip,) = spot.ilks(_ilk);
         require(_pip != address(0), "IlkRegistry/pip-invalid");
 
-        (address _flip,,) = cat.ilks(_ilk);
-        require(_flip != address(0), "IlkRegistry/flip-invalid");
-        require(FlipLike(_flip).cat() == address(cat), "IlkRegistry/flip-wrong-cat");
-        require(FlipLike(_flip).vat() == address(vat), "IlkRegistry/flip-wrong-vat");
+        (address _clip,,,) = dog.ilks(_ilk);
+        (address _flip,,)  = cat.ilks(_ilk);
+        address  _xlip;
+        uint128  _code;
+        if (_clip != address(0)) {
+            require(ClipLike(_clip).dog() == address(dog), "IlkRegistry/clip-wrong-dog");
+            require(ClipLike(_clip).vat() == address(vat), "IlkRegistry/clip-wrong-vat");
+            _xlip = _clip;
+            _code = 1;
+        } else if (_flip != address(0)) {
+            require(FlipLike(_flip).cat() == address(cat), "IlkRegistry/flip-wrong-cat");
+            require(FlipLike(_flip).vat() == address(vat), "IlkRegistry/flip-wrong-vat");
+            _xlip = _flip;
+            _code = 2;
+        } else {
+            revert("IlkRegistry/invalid-auction-contract");
+        }
 
```

Comment:
Maybe check that xlip is authed on the vat?

Reply:
I guess we can take care of that in the spell.

Reply:
The `add` function is still unauthed here, so we need the sanity checks to ensure an invalid ilk isn't getting in.